In [1]:
%run ./Formulas_exam2.ipynb

# Sumary of the formulas to use in quantitative methods 1
## For all formulas using an argument array, you have to transform the dataset doing that:
1. Copy the line from excel to python, if its a column in excel transpose that to a line.
1. Create a variable that you are going to atribute the dataset in python
1. Atribute the valor of the variable using the function transform_array(array), in "array" substitute the line from excel
1. Put ' at the begin and the end of the argument of the function transform_array, like transform_array('ctrl+v from the entire line on excel')
1. The code is like variable=transform_array('ctrl+v from the entire line on excel')
1. Use that "variable" as an argument for all functions that you want
### All proportion codes were developed using the conservative variance var=0.25

### Functions
1. transform_array(array)
* Copy a line from excel end put in the argument of function without changing,with quotes '' in the begin and in the end.
* if its a column, transpose the column on excel to a line.

2. conf_inter_p() or conf_inter_p(array)
* confidence interval for proportion

3. conf_inter() or conf_inter(array)
* confidence interval for the mean of population

4. number_samples_z()
* Calculate the number of samples to achieve a margin of error using z_score for proportion or continuous

5. number_samples_t()
* Calculate the number of samples to achieve a margin of error using t_score

6. conf_inter_chi2() or conf_inter_chi2(array)
* confidence interval for Variance

7. hypo_test_z() or hypo_test_z(array)
* Hypothesis test of the mean using z_score

8. hypo_test_t() or hypo_test_t(array)
* Hypothesis test of the mean using t_score

9. hypo_test_p_t(array)
* Hypothesis test of the proportion using z_score and transforming the dataset using a benchmark

10. hypo_test_p() or hypo_test_p(array)
* Hypothesis test of the proportion using z_score and the user giving the categorical dataset or not

11. hypo_test_chi2() or hypo_test_chi2(array)
*  Hypothesis test of the standard deviation using chi_square


# Exercises

## An investment manager promises a mean rate of return of 0.8% to his clients. Estimate the 95% C.I. for the mean rate of return offered by that manager. Can the client trust the manager’s offer?

n = 32
X = 0.76%
S = 0.26%

In [16]:
conf_inter() 

What is the mean of the sample in the database?0.0076
What is the number of elements in the database?32
What is the standard deviation of the sample in the database?0.0026
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
0.0066991625141727094 <Xmean< 0.00850083748582729


## What should be the sample size (n) required to estimate the mean rate of return with 95% of confidence, and a margin of error of 0.10?
  
S = 0.26

In [4]:
number_samples_z()

Digite if the case is proportion (s or n)n
What is the standard deviation of the sample in the database?0.26
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the margin of error (ME should be decimal and use "." being the separator) that you want?0.1
The number of samples needed is 26


## What should be the sample size (n) required to estimate the mean rate of return with 95% of confidence, and a margin of error of 0.05?
  
S = 0.26

In [5]:
number_samples_z()

Digite if the case is proportion (s or n)n
What is the standard deviation of the sample in the database?0.26
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the margin of error (ME should be decimal and use "." being the separator) that you want?0.05
The number of samples needed is 104


### 2018 poll conducted by NBC asked respondents if they usually wait for a little and save money before buying an expensive product instead of buying immediately and pay instalments with an interest rate. The responses by 230 people showed that 180 (yes) to wait for a little, against 50 (no) that buy immediately. Estimate a 95% confidence interval for the proportion of savers.

In [8]:
conf_inter_p()

What is the number of elements in the database?230
What is the number of p/(p/ should be decimal and use "." being the separator) that you want?0.7826086
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.05
0.718 <Xmean< 0.8472 , with significance being: 5.0 %


## What should the required sample size (n) be to estimate the proportion of savers with 95% of confidence, and a margin of error of 0.02 (2%)?

p/=0.783

In [7]:
number_samples_z()

Digite if the case is proportion (s or n)s
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the margin of error (ME should be decimal and use "." being the separator) that you want?0.02
The number of samples needed is 2401


### As an investment manager, you need to test whether the volume of shares in millions of dollars traded by NASDAQ is different from 368 million. The results of 36 consecutive days were collected and summarized below:

n 36 days  
Sample mean 372.5 mi  
S.D. 15 mi  
a = 5%  

In [19]:
hypo_test_z()

What is the number of elements in the database?36
What is the null hypothesis(Ho should be decimal and use "." being the separator)?368
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?<>
What is the value to analysis power?375
What is the mean of the sample in the database?372.5
What is the standard deviation of the sample in the database?15
Z_score is 1.8 critical z_score is 1.959963984540054 H0_LL critical is363.10009003864985 H0_UL critical is372.89990996135015
Not Reject null hypothesis p_value is 0.07186063822585154
The value of Beta is 0.20044409670188912
The value power is 0.7995559032981109


### As an investment manager, you need to test whether the volume of shares (in millions of dollars) traded by NASDAQ is greater than 368 million. The results of 36 consecutive days were collected and summarized below:
  
n 36 days  
Sample mean 372.5 mi  
S.D. 15 mi  
What is your conclusion?  
Consider a = 5%  

In [23]:
hypo_test_z()

What is the number of elements in the database?36
What is the null hypothesis(Ho should be decimal and use "." being the separator)?368
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?375
What is the mean of the sample in the database?372.5
What is the standard deviation of the sample in the database?15
Z_score is 1.8 critical z_score is 1.6448536269514722
Reject null hypothesis p_value is 0.03593031911292577
The value of Beta is 0.12401524544987613
The value of Power is 0.8759847545501238
Critical value of mu  372.1121340673787


## As an investment manager, you need to test whether the volume of shares in millions of dollars traded by NASDAQ is different from 368 million. The result of 25 consecutive days was collected and summarized below:
n 25 days  
Sample mean 373.3 mi  
S.D. 12.6 mi  
What is your conclusion?  
Consider a = 5%

In [30]:
hypo_test_t()

What is the null hypothesis(Ho should be decimal and use "." being the separator)?368
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the number of elements in the database?25
What is the alternative hypothesis(choose "<>" or "<" or ">")?<>
What is the value to analysis power?376
What is the mean of the sample in the database?373.3
What is the standard deviation of the sample in the database?12.6
T_score is 2.1031746031746077 critical T_score is 2.0638985616280205 H0_LL critical is362.7989756246974 H0_UL critical is373.2010243753026
Reject null hypothesis p_value is 0.046118054864621616
The value of Beta is 0.1388498969050262
The value power is 0.8611501030949738


## As an investment manager, you need to test whether the volume of shares in millions of dollars traded by NASDAQ is greater from 368 million. The result of 25 consecutive days was collected and summarized below:
n 25 days  
Sample mean 373.3 mi  
S.D. 12.6 mi  
What is your conclusion?  
Consider a = 5%

In [31]:
hypo_test_t()

What is the null hypothesis(Ho should be decimal and use "." being the separator)?368
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the number of elements in the database?25
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?376
What is the mean of the sample in the database?373.3
What is the standard deviation of the sample in the database?12.6
T_score is 2.1031746031746077 critical T_score is 1.7108820799094275
Reject null hypothesis p_value is 0.023059027432310808
The value of Beta is 0.07812286172521829
The value of Power is 0.9218771382747817
Critical value of mu  372.3114228413718


## As an investment manager, you need to test whether the risk of NASDAQ is different from 17 mi. The risk or volatility in an investment is measured by the standard deviation. The result of 25 consecutive days was collected and summarized below: NASDAQ

n 25 days  
Sample mean 373.3 mi  
S.D. 12.6 mi  
What is your conclusion?  
Consider a = 5%  

In [43]:
hypo_test_chi2()

What is the null hypothesis for standard deviation(Ho should be decimal and use "." being the separator)?17
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?<>
What is the number of elements in the database?25
What is the variance of the sample in the database?158.76
chi2_score is 13.184221453287197 critical Upper chi2_score is 39.36407702660391 critical Lower chi2_score is 12.401150217444435 "s" Upper critical is 21.771749911188934 "s" Lower critical is 12.220086628241502
Not Reject null hypothesis


##  The annual rate of return (ARoR) was calculated for each of 63 electric-power suppliers operating in 1995. The data were extracted from The Wall Street Journal (29th February 1996) and are available at worksheet ARoR - Database. As an investment manager, you should test the hypothesis that the ARoR mean of the electric-power suppliers is greater than 0.03. Use a = 5% significance . What is your conclusion?

In [23]:
x=transform_array('0,0524	0,0277	0,0348	0,0538	0,0303	0,0633	0,0444	0,0242	0,0590	0,0305	0,0406	0,0318	0,0284	0,0381	0,0392	0,0428	0,0210	0,0368	0,0319	0,0198	0,0320	0,0380	0,0375	0,0304	0,0359	0,0179	0,0184	0,0353	0,0259	0,0178	0,0327	0,0461	0,0255	0,0323	0,0246	0,0544	0,0226	0,0354	0,0208	0,0438	0,0318	0,0412	0,0278	0,0245	0,0257	0,0388	0,0329	0,0374	0,0420	0,0397	0,0248	0,0055	0,0324	0,0448	0,0445	0,0318	0,0234	0,0261	0,0354	0,0190	-0,0271	0,0086	0,0224')
hypo_test_z(x)

What is the null hypothesis(Ho should be decimal and use "." being the separator)?0.03
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?0.035
mean 0.03192857142857143
Standard deviation (s) 0.01346540883237644
Z_score is 1.1368062676320672 critical z_score is 1.6448536269514722
Not Reject null hypothesis p_value is 0.12780964132420658
The value of Beta is 0.09638610320089086
The value of Power is 0.9036138967991092
Critical value of mu  0.032790464654745735


In [4]:
#considering Ho<>
hypo_test_t(x)

What is the null hypothesis(Ho should be decimal and use "." being the separator)?0.03
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?0.035
mean 0.03192857142857143
Standard deviation (s) 0.01346540883237644
T_score is 1.1368062676320672 critical T_score is 1.669804162296528
Not Reject null hypothesis p_value is 0.1299970019746004
The value of Beta is 0.10309942790685186
The value of Power is 0.8969005720931481
Critical value of mu  0.03283279279012298


## The Renewable Energy Certificate (REC) is a market-based instrument that represents the property rights to the environmental, social and other non-power attributes of renewable electricity generation. The RECs are issued when one megawatt-hour(MWh) of electricity is generated and delivered to the electricity grid from a renewable energy resource. The Database worksheet presents the REC condition for each of 63 electric-power suppliers (REC=1 means that the company is certified, REC=0 means that the company is not certified). As an investment manager, you should test the hypothesis that the proportion of certified electric-power suppliers is less than 0.5. Use a=5% significance. What is your conclusion?

In [5]:
a=transform_array('1	1	0	0	1	0	0	0	1	0	0	1	0	0	0	0	0	0	0	1	0	1	1	0	0	1	0	0	1	1	1	0	0	0	0	1	1	1	0	1	0	1	1	0	0	0	1	1	0	0	0	1	0	0	0	1	1	1	1	0	1	0	0')
hypo_test_p(a)

What is the null hypothesis(Ho should be decimal and use "." being the separator)?0.5
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?<
Proportion is 0.4127
Z_score is -1.3858697343671669 critical z_score is -1.6448536269514729
Not Reject null hypothesis p_value is 0.08289334199052911


## The annual rate of return (ARoR) was calculated for each of 63 electric-power suppliers operating in 1995. The data were extracted from The Wall Street Journal (29th February 1996) and are available at worksheet ARoR - Database. The risk or volatility of the asset, is measured by the standard deviation. The Risk-free Investiment has a standard deviation of 0.0085. As an investment manager, you should test the hypothesis that the standard deviation of annual rate of return of shares of electricity companies is greater than 0.0085. Use a = 5% significance . What is your conclusion?

In [7]:
hypo_test_chi2(x)

What is the null hypothesis for standard deviation(Ho should be decimal and use "." being the separator)?0.0085
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
mean 0.03192857142857143
Standard deviation (s) 0.01346540883237644
Variance (s²) 0.00018131723502304146
chi2_score is 155.59402867029164 critical chi2_score is 81.3810151888991 "s" critical is0.009738333257567795
Reject null hypothesis


In [24]:
#considering Ho<>
hypo_test_chi2(x)

What is the null hypothesis for standard deviation(Ho should be decimal and use "." being the separator)?0.0085
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?<>
mean 0.03192857142857143
Standard deviation (s) 0.01346540883237644
Variance (s²) 0.00018131723502304146
chi2_score is 155.59402867029164 critical Upper chi2_score is 85.65373078961532 critical Lower chi2_score is 42.125991958283706 "s" Upper critical is 0.009990707012301004 "s" Lower critical is 0.00700645200233788
Reject null hypothesis


## A logistics consultant noted the time spent (in minutes) to process sixty orders during the operations of the Venda Virtual, a virtual sales company. For this sample, the mean time spent was 23.42 minutes with a standard deviation of 12.12 minutes.
(a) Define a 95% confidence interval for the mean, μ, the mean time spent (in minutes) to process orders during the
operations of the Venda Virtual.  
(b) Test the hypothesis that the mean μ is greater than 20, with a = 5% significance. What is your conclusion?  
(c) Calculate the test p-value.  
(d) Calculate the probability of error b for a mean μ = 25.  

In [2]:
#question a)
conf_inter()

What is the mean of the sample in the database?23.42
What is the number of elements in the database?60
What is the standard deviation of the sample in the database?12.12
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
20.353273219998858 <Xmean< 26.486726780001145


In [3]:
#question b),c),d)
hypo_test_z()

What is the number of elements in the database?60
What is the null hypothesis(Ho should be decimal and use "." being the separator)?20
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?25
What is the mean of the sample in the database?23.42
What is the standard deviation of the sample in the database?12.12
Z_score is 2.185743076572504 critical z_score is 1.6448536269514722
Reject null hypothesis p_value is 0.014417203723403715
The value of Beta is 0.06048953213938697
The value of Power is 0.939510467860613
Critical value of mu  22.573678244469296


## The mean number of customers each day at a gas station has been around 250. During a 25-day campaign offering customers a gift, the mean number of customers reached 280 with a standard deviation of 50. Would you say that the campaign altered the distribution of the number of customers at this gas station?
(a) Construct an adequate hypothesis test, at the 2% significance level, to help the manager. What is the decision? What
does this mean in practical terms?  
(b) Construct a 99% confidence interval for the mean number of customers each day during the campaign.  
(c) Now assume that the campaign lasted half a year, with a sample of 180 days and the sample average number of
customers at 280, with a sample standard deviation of 50. What would be the significance of the test with the following
critical region: C.R. = {𝑋 "ÎÂ/𝑋 " >258}?  

In [8]:
#a)
hypo_test_t()

What is the null hypothesis(Ho should be decimal and use "." being the separator)?250
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.02
What is the number of elements in the database?25
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?280
What is the mean of the sample in the database?280
What is the standard deviation of the sample in the database?50
T_score is 3.0 critical T_score is 2.1715446760080677
Reject null hypothesis p_value is 0.003102868308262341
The value of Beta is 0.20778790764589644
The value of Power is 0.7922120923541036
Critical value of mu  271.7154467600807


In [5]:
#b)
conf_inter()

What is the mean of the sample in the database?280
What is the number of elements in the database?25
What is the standard deviation of the sample in the database?50
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.01
252.03060495227197 <Xmean< 307.969395047728


In [7]:
#c)
alpha=(1-norm.cdf(x=258, loc=250, scale=50/math.sqrt(180)))*100
print(round(alpha,3),'%')

1.591 %


## The producers of a television program intend to change it if watched regularly by less than a quarter of the total audience. A poll conducted showed that 80 of the 400 families interviewed watched the program regularly.
(a) Construct an adequate hypothesis test, at the 6% significance level, to help the producers. What is their decision? What does this mean in practical terms?  
(b) Calculate the test description level. (p_value) 

In [9]:
hypo_test_p()

What is the null hypothesis(Ho should be decimal and use "." being the separator)?0.25
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.06
What is the alternative hypothesis(choose "<>" or "<" or ">")?<
What is the number of elements in the database?400
What is the number of p/(p/ should be decimal and use "." being the separator) that you want?0.2
Z_score is -1.9999999999999996 critical z_score is -1.5547735945968535
Reject null hypothesis p_value is 0.022750131948179216


## An investment manager promises a mean rate of return of 0.24% to his clients. A new strategy was defined to ensure a better mean rate of return for them. In the past, the standard deviation was around 0.039 and it should be kept. In order to check the hypotheses, eight days were selected at random, with the following rate of return: 0.242; 0.226; 0.210; 0.290;0.306; 0.322; 0.274 and 0.258.

(a) Construct an adequate hypothesis test, at the 10% significance level, to check the manager's promise in terms of the
customer view. What is its decision? What does this mean in practical terms?  
(b) Construct an adequate hypothesis test, at the 5% significance level, to check the manager's promise in terms of the
customer. What is its decision? What does this mean in practical terms?

In [15]:
a=transform_array('0,242	0,226	0,21	0,29	0,306	0,322	0,274	0,258')
# a)
hypo_test_t(a)

What is the null hypothesis(Ho should be decimal and use "." being the separator)?0.24
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.1
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?0.3
mean 0.266
Standard deviation (s) 0.03919183588453085
T_score is 1.8763883748662853 critical T_score is 1.4149239276488585
Reject null hypothesis p_value is 0.051354725604052964
The value of Beta is 0.011246585502118905
The value of Power is 0.9887534144978811
Critical value of mu  0.25960576105226185


In [14]:
# b)
hypo_test_t(a)

What is the null hypothesis(Ho should be decimal and use "." being the separator)?0.24
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?0.3
mean 0.266
Standard deviation (s) 0.03919183588453085
T_score is 1.8763883748662853 critical T_score is 1.894578605061305
Not Reject null hypothesis p_value is 0.051354725604052964
The value of Beta is 0.022526118669050743
The value of Power is 0.9774738813309493
Critical value of mu  0.2662520512231932


## The unit prices (in Real) of a specific part offered by five suppliers are 119.4; 114.6; 107.8; 111.0 and 111.5. The engineer in charge states that there was an illegal restatement and the mean price of the part now exceeds RS 110 (suggested price of the product):
(a) Construct an adequate hypothesis test, at the 5% level of significance, to evaluate the engineer's statement is true.  
(b) Assume that a sample was collected for 36 suppliers. This information showed a mean price of RS 112.5 and a standard
deviation of RS 5.00.Repeat item (a) based on the new information.  


In [18]:
# a)
b=transform_array('119,4	114,6	107,8	111,0	111,5')
hypo_test_t(b)

What is the null hypothesis(Ho should be decimal and use "." being the separator)?110
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?115
mean 112.85999999999999
Standard deviation (s) 4.379269345450222
T_score is 1.4603245224670909 critical T_score is 2.13184678133629
Not Reject null hypothesis p_value is 0.1089929402480706
The value of Beta is 0.3476390702376318
The value of Power is 0.6523609297623683
Critical value of mu  114.1751553855449


In [19]:
# b)
hypo_test_z()

What is the number of elements in the database?36
What is the null hypothesis(Ho should be decimal and use "." being the separator)?110
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?115
What is the mean of the sample in the database?112.5
What is the standard deviation of the sample in the database?5
Z_score is 3.0 critical z_score is 1.6448536269514722
Reject null hypothesis p_value is 0.0013498980316301035
The value of Beta is 6.648901509735581e-06
The value of Power is 0.9999933510984903
Critical value of mu  111.37071135579289


## A poll, conduct by NBC, showed that 85 out of 400 families interviewed in low-income communities had desktop computers in their homes. The Mayor will run a desktop computer purchasing loan program if the proportion of all residents with desktop computers is less than 30% (0.30).
(a) Construct a 98% confidence interval for the proportion of families living in low-income communities with desktop
computers. Based on the results, must the Mayor launch the loan program? Justify.  


In [3]:
#a)
conf_inter_p()

What is the number of elements in the database?400
What is the number of p/(p/ should be decimal and use "." being the separator) that you want?0.2125
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.02
0.1543 <Xmean< 0.2707 , with significance being: 2.0 %


## Listed below are the electricity consumption figures (in kWh) for the home of a resident of Rio de Janeiro for five different months: 1194, 1146, 1078, 1110 and 1115. In the last electricity bill, the supplier stated that fines will be imposed on residents with mean monthly consumption levels greater than 1150 kWh.
(a) Construct a 95% confidence interval for the mean monthly consumption of this resident in the suburbs of Rio de Janeiro.
Will the consumer under assessment have to pay a fine? Suggest an alternative approach in order to realize the test.  
(b) A new sample was collected for the last 36 months of consumption for this resident and showed X/= 1125 kWh and S=
50 kWh. Based on this new sample, construct a 95% confidence interval for the mean monthly consumption of this
resident. Will the consumer under assessment have to pay a fine?  
(c) Based on the information in item (b), what size must a new sample be for estimating mean electricity consumption with
an error of no more than 10 kWh and a 99% confidence interval?

In [6]:
#a)
c=transform_array('1194	1146	1078	1110	1115')
conf_inter(c)

The minimum value is  1078.0
The mean value is  1128.6
The variance value is  1917.7999999999997
The standard deviation is  43.7926934545022
The maximum is 1194.0
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
1074.2241891531708 <Xmean< 1182.975810846829


In [7]:
#b)
conf_inter()

What is the mean of the sample in the database?1125
What is the number of elements in the database?36
What is the standard deviation of the sample in the database?50
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.05
1108.6669667954995 <Xmean< 1141.3330332045005


In [8]:
#c)
number_samples_z()

Digite if the case is proportion (s or n)n
What is the standard deviation of the sample in the database?50
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.01
What is the margin of error (ME should be decimal and use "." being the separator) that you want?10
The number of samples needed is 166


## The label on a soft drink bottle made by Karioka Refrescos S.A. indicated that it contains 500ml of guaraná drink. However, the Consumer Protection Agency has received several complains that the company is delivering less guaraná drink than stated on the label. An independent consulting firm selected sixty bottles at random from assorted retailers nationwide, finding a mean of 489 ml and a standard deviation of 21 ml.
(a) Construct an adequate hypothesis test, at the 3% significance level, to check the complaints from consumers. What is
the decision? What does it mean in practical terms?  


In [9]:
#a)
hypo_test_z()

What is the number of elements in the database?60
What is the null hypothesis(Ho should be decimal and use "." being the separator)?500
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.03
What is the alternative hypothesis(choose "<>" or "<" or ">")?<
What is the value to analysis power?550
What is the mean of the sample in the database?489
What is the standard deviation of the sample in the database?21
Z_score is -4.0574111245982465 critical z_score is -1.880793608151251
Reject null hypothesis p_value is 2.4809845078521063e-05
The value of Beta is 3.9785758237794225e-92
The value of Power is 1.0
Critical value of mu  494.9010023745838


## An investment broker believes that the rate of return on its investments exceeds 24% a month. You wish to invest with this broker and analysed the rates of return (%) for the past 8 months finding: 23.6; 22.8; 25.7; 24.8; 26.4; 24.3; 23.9 and 25.0.Note: Rate of return (RR) is a metric variable that may range between −∞ < RR < ∞.
(a) Construct a hypothesis test, at the 10% significance level, to check the statement from the brokerage service. What is
your decision? Can you trust the brokerage service?  
(b) Calculate the significance test considering std= 1,1624, 121 months and C.R. = {X/>24.2}.
(c) Using the information in item (b), what is the power of the test for µ = 24.5%? What does it mean?

In [13]:
#a)
a=transform_array('23,6	22,8	25,7	24,8	26,4	24,3	23,9	25,0')
hypo_test_t(a)

What is the null hypothesis(Ho should be decimal and use "." being the separator)?24
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.1
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?24.5
mean 24.5625
Standard deviation (s) 1.1624327937562664
T_score is 1.3686728955130663 critical T_score is 1.4149239276488585
Not Reject null hypothesis p_value is 0.1067022111221001
The value of Beta is 0.4242135240695173
The power is 0.5757864759304827
Critical value of mu  24.581508344259372


In [15]:
#b)
hypo_test_z()

What is the number of elements in the database?121
What is the null hypothesis(Ho should be decimal and use "." being the separator)?24
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.0292
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the value to analysis power?24.5
What is the mean of the sample in the database?24.2
What is the standard deviation of the sample in the database?1.1624
Z_score is 1.8926359256710186 critical z_score is 1.892683289273909
Not Reject null hypothesis p_value is 0.029203151363872082
The value of Beta is 0.0022634207853957874
The value of Power is 0.9977365792146042
Critical value of mu  24.20000500504109


## The car insurance companies are thinking about raising premiums (amount charged to policyholders for insurance) for drivers using a mobile while driving. The National Consumer Group (NCG) affirms that the problem is not so serious, as only 10% of drivers using a mobile while driving. An independent consulting conducted a study with a random sample of drivers. It was noted that 60 out of 500 drivers were using the mobile while they were driving.
(a) Construct an adequate hypothesis test, at the 2% significance level, to help NCG (consider NCG is the status quo). What
is your decision? What does it mean in practical terms?   
(b) How many drivers should be investigated to estimate a 96% confidence interval with an error maximum of 2%
(conservative)?

In [16]:
#a)
hypo_test_p()

What is the null hypothesis(Ho should be decimal and use "." being the separator)?0.1
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.02
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the number of elements in the database?500
What is the number of p/(p/ should be decimal and use "." being the separator) that you want?0.12
Z_score is 0.8944271909999154 critical z_score is 2.0537489106318225 H0 critical is0.14592322172888897
Not Reject null hypothesis p_value is 0.18554668476134895


In [18]:
#b)
number_samples_z()

Digite if the case is proportion (s or n)s
What is the number of the significance(alpha should be decimal and use "." being the separator) that you want?0.04
What is the margin of error (ME should be decimal and use "." being the separator) that you want?0.02
The number of samples needed is 2637


## The financial market calls "conservative" an investment considered of low risk. The risk or volatility of the asset is measured by the standard deviation. An agency considers as a conservative investment that whose risk does not exceed 0.0740 (7.4%). The BOVESPA index in the period from Abr08 to Fev14 (71 months) presented an average rate of return of 0.0158 and standard deviation of 0.0788. By the BOVESPA index's historical data knows that its distribution can be approximated by a normal distribution.
(a) (2,0) Conduct an appropriate hypotheses test, at the 10% level of significance, to verify if there is evidence that the
BOVESPA index is a non-conservative investment.

In [20]:
print('The variance is ',0.0788**2)

The variance is  0.006209439999999999


In [21]:
hypo_test_chi2()

What is the null hypothesis for standard deviation(Ho should be decimal and use "." being the separator)?0.074
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.1
What is the alternative hypothesis(choose "<>" or "<" or ">")?>
What is the number of elements in the database?71
What is the variance of the sample in the database?0.0062094
chi2_score is 79.37509130752376 critical chi2_score is 85.52704271487188 "s" critical is0.08179644472771927
Not Reject null hypothesis


## A market researcher for an automobile dealer intends to conduct a nationwide survey concerning car repairs. Among the questions included in the survey is the following: “What was the cost of all repairs performed on your car last year?” In order to determine the sample size necessary, the researcher needs to provide an estimate of the standard deviation. Using his past experience and judgment, he estimates that the standard deviation of the amount of repairs is S 200. Suppose that a smallscale study of 25 auto owners selected at random indicates a sample standard deviation of S237.52.
(a) At the 0.05 level of significance, is there evidence that the population standard deviation is different from $200?  
(b) What assumption do you need to make in order to perform this test?

In [23]:
print('The variance is ',237.52**2)

The variance is  56415.750400000004


In [24]:
#a)
hypo_test_chi2()


What is the null hypothesis for standard deviation(Ho should be decimal and use "." being the separator)?200
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?<>
What is the number of elements in the database?25
What is the variance of the sample in the database?56415.7504
chi2_score is 33.84945024 critical Upper chi2_score is 39.36407702660391 critical Lower chi2_score is 12.401150217444435 "s" Upper critical is 256.1382342492816 "s" Lower critical is 143.76572503813532
Not Reject null hypothesis


In [ ]:
# b)
# The population of the variable X should follow a normal distribution

## The marketing manager of a branch office of a local telephone operating company wants to study characteristics of residential customers served by her office. In particular, she wants to estimate the mean monthly cost of calls within the local calling region. In order to determine the sample size necessary, she needs an estimate of the standard deviation. On the basis of her past experience and judgment, she estimates that the standard deviation is equal to S12. Suppose that a small-scale study of 15 residential customers indicates a sample standard deviation of S9.25.
(a) At the 0.10 level of significance, is there evidence that the population standard deviation is different from $12?  
(b) What assumption do you need to make in order to perform this test?

In [25]:
print('The variance is ',9.25**2)

The variance is  85.5625


In [26]:
#a)
hypo_test_chi2()

What is the null hypothesis for standard deviation(Ho should be decimal and use "." being the separator)?12
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.1
What is the alternative hypothesis(choose "<>" or "<" or ">")?<>
What is the number of elements in the database?15
What is the variance of the sample in the database?85.5625
chi2_score is 8.31857638888889 critical Upper chi2_score is 23.684791304840576 critical Lower chi2_score is 6.57063138378934 "s" Upper critical is 15.6081708178236 "s" Lower critical is 8.22092677807099
Not Reject null hypothesis


In [ ]:
# b)
# The population of the variable X should follow a normal distribution

## A manufacturer of doorknobs has a production process that is designed to provide a doorknob with a target diameter of 2.5 inches. In the past, the standard deviation of the diameter has been 0.035 inch. In an effort to reduce the variation in the process, various studies have resulted in a redesigned process. A sample of 25 doorknobs produced under the new process indicates a sample standard deviation of 0.025 inch.
(a) At the 0.05 level of significance, is there evidence that the population standard deviation is less than 0.035 inch in the new process?  
(b) What assumption do you need to make in order to perform this test?

In [27]:
print('The variance is ',0.025**2)

The variance is  0.0006250000000000001


In [29]:
#a)
hypo_test_chi2()

What is the null hypothesis for standard deviation(Ho should be decimal and use "." being the separator)?0.035
What is the number of the significance(alfa should be decimal and use "." being the separator) that you want?0.05
What is the alternative hypothesis(choose "<>" or "<" or ">")?<
What is the number of elements in the database?25
What is the variance of the sample in the database?0.000625
chi2_score is 12.244897959183671 critical chi2_score is 13.848425027170224 "s" critical is0.026586588613343138
Reject null hypothesis
